navbar empty when state's data object has incorrect up property

davidpfahler opened the pull request:
When a state has an `up` property defined on its `data` object and the specified state does not exist, the script breaks and hence the navbar fails to compile (and appears empty).

```
$stateProvider.state('app', {
  url: '/',
  templateUrl: 'table.html',
  data: {
    up: 'asdfgh'
  }
});
```

```
TypeError: Cannot read property 'data' of null
    at getTitleFromState (http://localhost:9000/bradypodion.js:196:37)
    at http://localhost:9000/bradypodion.js:234:34
    at publicLinkFn (http://localhost:9000/bower_components/angular/angular.js:5586:29)
    at http://localhost:9000/bradypodion.js:213:11
    at nodeLinkFn (http://localhost:9000/bower_components/angular/angular.js:6271:13)
    at compositeLinkFn (http://localhost:9000/bower_components/angular/angular.js:5678:15)
    at publicLinkFn (http://localhost:9000/bower_components/angular/angular.js:5587:30)
    at http://localhost:9000/bradypodion.js:339:32
    at Scope.$broadcast (http://localhost:9000/bower_components/angular/angular.js:12329:28)
    at $state.transition.resolved.then.$state.transition (http://localhost:9000/bower_components/angular-ui-router/release/angular-ui-router.js:1822:22) <bp-navbar class="ng-scope" role="navigation"> 
```

![screen shot 2014-03-15 at 20 57 18](https://f.cloud.github.com/assets/908242/2429349/0ec94fb6-ac7c-11e3-916f-1d21dd9ceae3.png)

Problem is that the [detected upState is not checked for existence](https://github.com/excellenteasy/bradypodion/blob/master/modules/scripts/directives/navbar.js#L129).

davidnpma committed:
test(navbar): should not throw if up state does not exist

davidnpma committed:
fix(navbar): only compile up button if up state is found

davidpfahler commented:
@boennemann This is an attempt to copy with the issue described above. I'm still new to angular error handling and not sure if this is the right approach. It is definitely more descriptive than "Cannot read property 'data' of null" error, but I would like to give better directions on where to fix / where code is broken, which is not clear from the stack trace. You can play around with this easily by giving a state on the demo app a false data.up property.

Also, it might be worth discussing whether this should fail or not. We could also just discard the up state and button and continue with the navbar rendering.

davidnpma committed:
test(view): direction detection should work with params in state url

Added three expectations regarding the direction detection when the
state url has a parameter in the state URL which is not the last URL segment.

boennemann commented:
Rather than throwing an error I'd just compile the navbar without the up button, and use `$log.warn` or `$log.error` to inform the user. http://docs.angularjs.org/api/ng/service/$log

davidnpma committed:
fix(navbar): log error when param in state url

Parameters in URL currently prevent us from detection the up state
automatically. Hence, we remind the user the they need to set the
up property on the data object in their state's configuration.
